src/core/compact.rs

```diff
@@ -85,153 +89,172 @@ pub trait FindexCompact<
         // We need to fetch all the Entry Table to re-encrypt it.
         // First, fetch all UIds of the Entry table
         let all_uids = self.fetch_all_entry_table_uids().await?;
-        let encrypted_entry_table = self.fetch_entry_table(&all_uids).await?;
-
-        // The goal of this function is to build these two data sets (along with
-        // `chain_table_uids_to_remove`) and send them to the callback to update the
-        // database.
-        let mut chain_table_adds = EncryptedTable::default();
-        let mut entry_table = EntryTable::decrypt::<BLOCK_LENGTH, DEM_KEY_LENGTH, DemScheme>(
-            &k_value,
-            &encrypted_entry_table,
-        )?;
-        // Entry Table items indexing empty chains should be removed.
-        let mut entry_table_uids_to_drop = Vec::new();
+        let all_uids_vec = Vec::from_iter(all_uids.clone());
 
         // Randomly select `n_lines` Entry Table lines such that an average number of
         // `num_reindexing_before_full_set` calls to `compact` are needed to compact all
         // the chain table.  See `documentation/Findex.pdf` and coupon problem to
         // understand this formula.
-        let entry_table_length = entry_table.len() as f64;
+        let entry_table_length = all_uids.len() as f64;
         let n_lines = ((entry_table_length * (entry_table_length.log2() + 0.58))
             / f64::from(num_reindexing_before_full_set))
         .ceil() as usize;
-        let entry_table_items_to_reindex = entry_table
-            .keys()
+        let entry_table_uids_to_reindex = all_uids
+            .iter()
             .choose_multiple(&mut rng, n_lines)
             .into_iter()
             .cloned()
-            .collect::<Vec<_>>();
+            .collect::<HashSet<_>>();
```

Comment:
You shouldn't need a `HashSet` here

Reply:
The hashset is used for computing an intersection